### PR TITLE
added K8S services for proxy tunnel mode

### PR DESCRIPTION
**Pull request by sa-ChristianAnton**
When using tunnel mode "host" and specifying services to be published with an opened TCP port by this router, with this PR a "default" service is automatically created which holds all these proxy tunnel service ports.

Additionally, it is possible to distribute several ZitiServices and their listen ports to different Kubernetes services by adding additional K8s services and give them short names. Use cases are those where you want, for example, one zitiService to be published on a ClusterIP service, when another should be on a LoadBalancer or NodePort service.

### CODE REVIEW

File: charts/ziti-router/values.yaml [hunk 1]
```diff
@@ -82,8 +82,24 @@ tunnel:
...
+  # -- Kubernetes services created additionally of the "default" proxy listener service if mode "proxy"
+  additionalK8sServices: []
+    #- name: myservice
+    #  type: ClusterIP
+  # -- list of definitions to configure proxy services and according kubernetes services for these if mode "proxy"
+  proxyServices: []
+    #  # -- name of the Openziti service to forward data to
+    #- zitiService: my_service.svc
+    #  # -- port the router opens to listen for data
+    #  listenPort: 8443
```
[the remark this answers is not in the document]
Comment (qrkourier) [reply charts/ziti-router/values.yaml:98]: Additionally, the possibility of a "serviceNodePort" is introduced in the service template, but we could let the advertised port serve this purpose when type=NodePort. That will avoid the need to introduce a another optional value "serviceNodePort." If you prefer to parameterize "advertisedPort" and "serviceNodePort" separately, then please add a comment to document "serviceNodePort."

**Comment by sa-ChristianAnton** [comment]
Hope everything is fine now :)

**Comment by qrkourier** [comment]
Hi @sa-ChristianAnton, I am working to test your branch, and I encountered a router error message. I don't know yet whether this error is caused by some issue we need to address in this branch or if it might be my mistake or misunderstanding. I'll share the link here just in case you recognize the message "[invalid edge router for session](https://openziti.discourse.group/t/invalid-edge-router-for-session/1208?u=qrkourier)."

**Comment by qrkourier** [comment]
> @sa-ChristianAnton Nice work on this. I was able to test it successfully. What do you think about creating Ingress resources for the services? Is it a good idea? In your environment, did you configure those separately? It's a bit complex because we will eventually want to support Ingress and Gateway+VirtualService, but we could start with Ingress.

I wrote this Terraform module to demonstrate using the router as a proxy: https://github.com/openziti-test-kitchen/terraform-lke-ziti/blob/add-router-proxy/plan-70-router-proxy/main.tf#L85

**Comment by sa-ChristianAnton** [comment]
> What do you think about creating Ingress resources for the services?

That would actually probably have been my next pull request. I have thought about it, but I currently, in my setup, don't need Ingresses for those.

**Pull request merged**